Stripe/STPCardValidator.m

```diff
@@ -111,19 +111,21 @@ + (STPCardValidationState)validationStateForNumber:(nonnull NSString *)cardNumbe
     if (![self stringIsNumeric:sanitizedNumber]) {
         return STPCardValidationStateInvalid;
     }
-    
     NSArray *brands = [self possibleBrandsForNumber:sanitizedNumber];
     if (brands.count == 0 && validatingCardBrand) {
         return STPCardValidationStateInvalid;
     } else if (brands.count >= 2) {
         return STPCardValidationStateIncomplete;
     } else {
         STPCardBrand brand = (STPCardBrand)[brands.firstObject integerValue];
-        NSInteger desiredLength = [self lengthForCardBrand:brand];
-        if ((NSInteger)sanitizedNumber.length > desiredLength) {
+        NSArray<NSNumber *>* lengths = [self lengthsForCardBrand:brand];
+        NSUInteger maxLength = [[lengths lastObject] integerValue];
+        NSUInteger length = sanitizedNumber.length;
+        if (length > maxLength) {
             return STPCardValidationStateInvalid;
-        } else if ((NSInteger)sanitizedNumber.length == desiredLength) {
-            return [self stringIsValidLuhn:sanitizedNumber] ? STPCardValidationStateValid : STPCardValidationStateInvalid;
+        } else if ([lengths containsObject:[NSNumber numberWithInteger:length]]) {
+            STPCardValidationState elseState = (length == maxLength) ? STPCardValidationStateInvalid : STPCardValidationStateIncomplete;
```

Comment:
I think this is an overly-terse ternary, probably better served by just another if/else.